### PR TITLE
Added pyproject.toml.

### PR DESCRIPTION
**Pull request by benjaminrwilson**
# Overview

This PR adds a `pyproject.toml` file to consolidate styling tools such as `isort` and `black`.

- `pyproject.toml` was introduced in [PEP 518](https://www.python.org/dev/peps/pep-0518/) for build system requirements.
- Allows for a unified configuration file for a host of tools, e.g. `isort`, `black`, etc.

Each "block" within this file specifies a different type of options. For example, `isort` is written as:

```
[tool.isort]

known_first_party = "argoverse"
line_length = 120
multi_line_output = 3
profile = "black"
```

The options above should have a 1-to-1 correspondence with `isort.cfg`. 

# Advantages

- Modern, consolidated project configuration
-  Lays groundwork for [poetry](https://python-poetry.org/) integration.

### CODE REVIEW

**Review by wqi — APPROVED** [review]
LGTM - thanks for separating the changes out from the previous PR.

**Comment by benjaminrwilson** [comment]
Removed `.isort.cfg` since `pyproject.toml` contains the configuration now.

**Comment by johnwlambert** [comment]
Hi Ben, does the .toml file change our pre-commit config, or our Tox configuration?

**Comment by benjaminrwilson** [comment]
None of that should change with this commit. `pre-commit` doesn't have support (although, it's widely requested) and `tox` is still defined in a separate file. I suggest `nox` support in a separate PR.

**Comment by johnwlambert** [comment]
isort states that .isort.cfg or pyproject.toml are both supported: https://pycqa.github.io/isort/docs/configuration/options/

File: .isort.cfg [hunk 1]
```diff
@@ -1,5 +0,0 @@
-[settings]
-known_first_party=argoverse
-# cat setup.py | grep -Pzo "(?s)install_requires.+?]" | tail -n+2 | head -n-1 | grep -Po '"[a-zA-Z0-9]+' | sed 's/"/,/'
-known_third_party=mayavi,pytest,colour,descartes,imageio,matplotlib,motmetrics,numpy,cv2,pandas,pillow,imageio,pyntcloud,scipy,shapely,sklearn
```
Comment (tagarwal-argoai) [inline .isort.cfg:4]: I believe this doesn't affect the installation of these packages, right?

**Review by tagarwal-argoai — APPROVED** [review]
LGTM!

File: .isort.cfg (hunk 1 above)
Comment (benjaminrwilson) [reply .isort.cfg:4]: That's right. I added them back in for now.

**Pull request merged**